sygnal/apnspushkin.py

```diff
@@ -128,6 +136,20 @@ def __init__(self, name, sygnal, config):
         # without this, aioapns will retry every second forever.
         self.apns_client.pool.max_connection_attempts = 3
 
+    def _report_certificate_expiration(self, certfile):
+        """Export the epoch time that the certificate expires as a metric."""
+        with open(certfile, "rb") as f:
+            cert = f.read()
+
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        # Convert from a string to a datetime object.
+        expiration_date = datetime.strptime(
+            x509.get_notAfter().decode(), "%Y%m%d%H%M%SZ"
+        )
```

Comment:
my impression (possibly unfounded) is that generally it's preferred to use `pyca/cryptography` rather than doing battle with openssl directly. https://cryptography.io/en/latest/x509/reference/#loading-certificates makes it look fairly easy. WDYT?

Reply:
If that's preferred we can do it that way, looks like it is pretty much the same.

For the record I used OpenSSL directly here since this is what aioapns uses (and I had code which did most of what I needed already).

Reply:
Much to my surprise.... pyOpenSSL requires cryptography...